### PR TITLE
Use all available cores

### PR DESCRIPTION
**Pull request by markbeierl**
When determining what cores are available, we should use the total number from the system, not from the current process

Addresses #780

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

File: conf/up4.bess [hunk 1]
```diff
@@ -99,7 +100,7 @@ else:
 
 
 # Initialize workers
-cores = get_process_affinity()
```
Comment (gab-arrobo) [inline conf/up4.bess:102]: Given that with this change the `get_process_affinity` function is not used anymore, should it be removed from `utils.py`?

Comment (markbeierl) [reply conf/up4.bess:102]: Excellent point. I was not sure about if it could have been used elsewhere without me knowing it.

File: conf/utils.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def getpythonpid(process_name):
 
 def get_json_conf(path, dump):
     try:
-        with open(path, 'r') as f:
+        with open(path, "r") as f:
```
Comment (markbeierl) [inline conf/utils.py:42]: Sorry about this - automatic formatter. Shall I revert?

File: conf/up4.bess (hunk 1 above)
Comment (gab-arrobo) [reply conf/up4.bess:102]: I do not see it used anywhere else

File: conf/utils.py (hunk 2 above)
Comment (gab-arrobo) [reply conf/utils.py:42]: No, please leave it. I applied formatting to the whole Python code a while ago (#748) but it looks like I unintentionally introduced the single quotes in a later PR

**Comment by gab-arrobo** [comment]
@markbeierl, have you tested the behavior of this PR when deploying multiple UPF instances in the same host?

**Comment by markbeierl** [comment]
Yes, I have tested it, and repeated tests show my affinity staying at `[1-15]` instead of increasing with subsequent runs.

**Comment by gab-arrobo** [comment]
@sureshmarikkannu, please review this PR. Thanks!

**Comment by gab-arrobo** [comment]
@markbeierl, you may need to rebase your PR and make the commit(s) appending the `-s` flag to pass the `DCO` check

**Comment by sureshmarikkannu** [comment]
@markbeierl , Sorry for the delayed review. I want to test multiple scenarios to understand the impact of this change and also want to understand the UPF behavior when we deploy multiple instances of UPF. 
Please find my observations below,
1. performance tests with single UPF instance: I run TCP/UDP tests with this patch and all looks good, no issues observed.
2. performance tests with multiple UPF instance: I do see that the same core was assigned for both UPF instances (bess worker threads) which impacts the throughput. Also please note that i do see same behavior without this patch as well (which means psutil.Process().cpu_affinity() returns the entire set of CPU's available in the system). 

So in summary when we scale the UPF, we need to utilize the core's carefully in order to not intersect with other UPF's. 
We need to use specific set of cores per instance (maybe based on the cores allocated for the UPF pod) for allocating the worker threads.

@gab-arrobo, @markbeierl  Let me know your thoughts.

**Comment by gab-arrobo** [comment]
close as explained in #780
